src/bundle/Resources/public/js/scripts/admin.context.menu.js

```diff
@@ -21,40 +25,109 @@
         },
         onAdapted: (visibleItems, hiddenItems) => {
             const hiddenButtonsIds = [...hiddenItems].map((item) => item.querySelector('.ibexa-btn').id);
+            const topBranchItems = multilevelPopupMenu.getBranchItems(topBranch);
 
-            popupMenu.toggleItems((popupMenuItem) => !hiddenButtonsIds.includes(popupMenuItem.dataset.relatedButtonId));
-        },
-    });
-    const popupMenu = new ibexa.core.PopupMenu({
-        popupMenuElement,
-        triggerElement: showPopupButton,
-        onItemClick: (event) => {
-            const { relatedButtonId } = event.currentTarget.dataset;
-            const button = doc.getElementById(relatedButtonId);
+            topBranchItems.forEach((branchItem) => {
+                const shouldBeVisible = hiddenButtonsIds.includes(branchItem.dataset.relatedBtnId);
 
-            button.click();
+                multilevelPopupMenu.toggleItemVisibility(branchItem, shouldBeVisible);
+            });
         },
     });
-    const popupItemsToGenerate = [...menuButtons].map((button) => {
-        const relatedButtonId = button.id;
-        const label = button.querySelector('.ibexa-btn__label').textContent;
+    const clickRelatedBtn = (relatedBtnId) => {
+        const button = doc.getElementById(relatedBtnId);
```

Comment:
btn? I'm lost if we use btn everywhere or not so maybe not

Reply:
Maybe
```suggestion
        const relatedBtnById = doc.getElementById(relatedBtnId);
```